### PR TITLE
Do not permit type parameters on builtin bounds

### LINKED ISSUE #20302: We permit type parameters and so forth on builtin bounds
Example:

```
fn foo<T:Copy<U>, U>(x: T) {
}

fn main() {
}
```

I left a FIXME at the relevant part of the code.

### PR DESCRIPTION
**Pull request by flaper87**
Fixes #20302

### CODE REVIEW

File: src/libtest/stats.rs [hunk 1]
```diff
@@ -332,6 +332,7 @@ pub fn winsorize<T: Float + FromPrimitive>(samples: &mut [T], pct: T) {
 
 /// Returns a HashMap with the number of occurrences of every element in the
 /// sequence that the iterator exposes.
+#[cfg(not(stage0))]
```
Comment (nikomatsakis) [inline src/libtest/stats.rs]: why is this `#[cfg(not(stage0))]`?

Comment (FlaPer87) [inline src/libtest/stats.rs]: for that bug in libtest after the `Hash` definition was changed. I've the same fix in the oibit patch, I guess we can wait for that one to land and remove this one.

**Comment by Manishearth** [comment]
Needs https://github.com/Manishearth/rust/commit/c94e4364245802867cdd05a456fb86dc46cf13c8 to compile on master.

**Pull request merged**